proof/exp.saw

```diff
@@ -124,25 +124,30 @@ let ovs_for_reciprocal =
 // an instance of (fp_exp_fact):
 recip_fp_algebra_thm <- admit_cryptol {{ \x -> inverse_fp x == fp_exp x (`p-2) }};
 
-let reciprocal_fp_spec = do {
+let reciprocal_fp_spec use_div_form = do {
     out_ptr <- crucible_alloc vec384_type;
     (inp, inp_ptr) <- ptr_to_fresh_readonly "inp" vec384_type;
     crucible_precond {{ fp_invariant inp }};
     crucible_execute_func [out_ptr, inp_ptr];
     // crucible_points_to out_ptr (crucible_term {{ fp_rep (fp_exp (fp_abs inp) (`p-2)) }});
-    // crucible_points_to out_ptr (crucible_term {{ fp_rep (Fp.div (Fp.field_unit, fp_abs inp)) }});
-    crucible_points_to out_ptr (crucible_term {{ fp_rep (inverse_fp (fp_abs inp)) }});
+    if use_div_form then do {
+        crucible_points_to out_ptr (crucible_term {{ fp_rep (Fp.div (Fp.field_unit, fp_abs inp)) }});
+    } else do {
+        crucible_points_to out_ptr (crucible_term {{ fp_rep (inverse_fp (fp_abs inp)) }});
+    };
     };
 
 reciprocal_fp_ov <- custom_verify "reciprocal_fp"
                     ovs_for_reciprocal
-                    reciprocal_fp_spec
+                    (reciprocal_fp_spec false)
                     do {
                       simplify (addsimp recip_fp_algebra_thm empty_ss);
                       unfolding ["sqr_n_mul"];
                       simplify fp_exp_simps;
                       w4_unint_z3 (concat ["fp_exp"] fp_unints);
                       };
+
+reciprocal_fp_div_ov <- admit "reciprocal_fp" (reciprocal_fp_spec true);
```

Comment:
I asked about this on mattermost, but want to call specific attention to it here too.  Is there an existing proof for the `Fp.div` postcondition in `reciprocal_fp_spec`?  Looking at the Git history it appears that it used to be the postcondition before the `inverse_fp` version was added.